File: Services/Tests Service/src/main/java/com/futurealgos/micro/testing/services/SearchService.java
/*******************************************************************************
 * Future Algorithms Private Limited
 * Copyright 2022 Future Algorithms Private Limited.
 * Unauthorized copying of any files, via any medium is strictly prohibited.
 * All Rights Reserved.
 ******************************************************************************/

package com.futurealgos.micro.testing.services;


import com.futurealgos.micro.testing.dto.payload.search.SearchDTO;
import com.futurealgos.micro.testing.models.root.BaseEntity;
import com.futurealgos.micro.testing.repos.SearchInterfaceRepository;
import com.futurealgos.micro.testing.repos.SearchRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

@Service
public class SearchService<T extends BaseEntity> {

    @Autowired
    SearchRepository searchRepository;

    @Autowired
    SearchInterfaceRepository searchInterfaceRepository;

    public List<T> search(String tableName, SearchDTO payload) {

        List<T> list = new ArrayList<>();
        Map<String, String> map = new HashMap<>();
        try {
            Class tclass = Class.forName(tableName);

//            list=searchRepository.search(tclass,payload.fields(),payload.searchFieldsDTOList());

            for (T t : list) {
                System.out.println(t);
            }

        } catch (Exception e) {
            e.printStackTrace();
            //throw new ClassNotFoundException("class "+ tableName +" not found");
        }

        return list;
    }

}

File: User Interface/src/app/layout/components/navbar/navbar-search/search.service.ts
/*
 * Future Algorithms Private Limited
 * Copyright 2022 Future Algorithms Private Limited.
 * Unauthorized copying of any files, via any medium is strictly prohibited.
 * All Rights Reserved.
 */

import {HttpClient} from '@angular/common/http';
import {Injectable} from '@angular/core';

import {BehaviorSubject} from 'rxjs';

@Injectable({
    providedIn: 'root'
})
export class SearchService {
    // Public
    public search = '';
    public apiData = [];
    public onApiDataChange: BehaviorSubject<any>;
    public onIsBookmarkOpenChange: BehaviorSubject<any>;

    /**
     *
     * @param {HttpClient} _httpClient
     */
    constructor(private _httpClient: HttpClient) {
        this.onApiDataChange = new BehaviorSubject('');
        this.onIsBookmarkOpenChange = new BehaviorSubject(false);
        this.getSearchData();
    }

    /**
     * Get Search Data
     */
    getSearchData(): Promise<any[]> {
        return new Promise((resolve, reject) => {
            this.apiData = [];
            this.onApiDataChange.next(this.apiData);
            resolve(this.apiData);
            // this._httpClient.get('api/search-data').subscribe((response: any) => {
            //
            // }, reject);
        });
    }
}
